wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/urlfactory/URLFactoryBuilder.java

```diff
@@ -84,11 +88,12 @@ public URLFactoryBuilder(
    * Build a factory using the provided devfile
    *
    * @param devfileLocation location of devfile
+   * @param devfileFileName name of devfile file
    * @param fileContentProvider service-specific devfile related file content provider
    * @return a factory or null if devfile is not found
    */
   public Optional<FactoryDto> createFactoryFromDevfile(
-      String devfileLocation, FileContentProvider fileContentProvider)
+      String devfileLocation, String devfileFileName, FileContentProvider fileContentProvider)
```

Comment:
When is the `devfileFileName` different from the last path segment of `devfileLocation`?